source/configuration_manual/mail_location/index.rst

```diff
@@ -1,299 +1,367 @@
+Variables
+---------
 
-SUBSCRIPTIONS       Specifies the file used for storing subscriptions. The default is ``subscriptions``. If you're trying to avoid name collisions with a mailbox named ``subscriptions``, then also consider setting ``MAILBOXDIR``.
+You can use several variables in the :ref:`setting-mail_location` setting.
+See :ref:`config_variables` for a full list, but the most commonly used ones
+are:
 
-MAILBOXDIR          Specifies directory name under which all mailbox directories are stored. With :ref:`dbox formats <dbox_mbox_format>` the default is ``mailboxes/`` while with other mailbox formats the default is empty. Typically this should be changed only for :ref:`lazy_expunge_plugin` and :ref:`namespaces` with mdbox.
+========= ==================================================================
+Variable  Description
+========= ==================================================================
+``%u``    Full username.
 
-DIRNAME             Specifies the directory name used for mailbox directories, or in the case of mbox specifies the mailbox message file name. With :ref:`dbox formats <dbox_mbox_format>` the default is ``dbox-Mails/`` while with other mailbox formats the default is empty. Can be used under either :ref:`mbox <mbox_settings_message_filename>`, `Maildir <maildir_settings_mailbox_directory_name>` , or :ref:`dbox <dbox_settings_mailbox_directory_name>` formats.
+``%n``    User part in ``user@domain``; same as ``%u`` if there's no domain.
+   
+``%d``    Domain part in ``user@domain``; empty if there's no domain.
+========= ==================================================================
 
-FULLDIRNAME         Same as ``DIRNAME``, but use the directory name also for index and control directory paths. This should be used instead of ``DIRNAME`` for new installations. (v2.2.8+)
+Directory Hashing
+^^^^^^^^^^^^^^^^^
 
-ALT                 specifies the `alternate storage <dbox_settings_alt_storage>` path for dbox formats.
-================= ==============================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================================
+You can use two different kinds of hashes in :ref:`config_variables`.
 
-* The colons and equals signs are literal and there are no spaces in an actual mailbox location specification.
+* ``%H`` returns a 32bit hash of the given string as hex.
```

Comment:
There are actually 3 different hashes:
 * %N is MD5-based "new hash" which works similarly to %H, except gives more uniform results.
 * %H is the old deprecated hash.
 * %M is the MD5 hash.